src/libstd/collections/hash/table.rs

```diff
@@ -938,29 +923,39 @@ impl<'a, K, V> Clone for RawBuckets<'a, K, V> {
     fn clone(&self) -> RawBuckets<'a, K, V> {
         RawBuckets {
             raw: self.raw,
-            hashes_end: self.hashes_end,
+            elems_left: self.elems_left,
             marker: marker::PhantomData,
         }
     }
 }
 
 
 impl<'a, K, V> Iterator for RawBuckets<'a, K, V> {
-    type Item = RawBucket<K, V>;
+    type Item = (*mut HashUint, *mut (K, V));
```

Comment:
It seems odd to have an iterator named `RawBuckets` that doesn't return `RawBucket`s.